File: Algorithms/Day 1 Linear Search/generateTestData.py
#To generate random integer test data as specified by input and save to a file
import random
import time

amountOfData = int(input("How many examples to generate?"))
min = int(input("Minimum val:"))
max = int(input("Maximum val:"))
separator = raw_input("What to separate with (type newline for \\n)?")
fileName = "testData" + str(time.time()).replace(".","") + ".txt"
f = open(fileName,"w")
for i in range(amountOfData):
	f.write(str(random.randint(min,max)))
	if separator == "newline":
		f.write("\n") 
	else:	
		f.write(separator)
f.close()
print(str(amountOfData)+" examples from " + str(min) + " to " + str(max) + ", separated by " + separator + " generated and saved to file " + fileName)

File: Algorithms/Day 1 Linear Search/generateTestData.c
//Doesn't quite work. Maybe you guys can help

#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <string.h>

int main () {
	FILE *myFile;
	char *fileName;
	time_t startTime;
	unsigned long numOfTestCases;
	int i;
	unsigned long max;
//	char separator[1];
	char separator = '\n';
	fileName = malloc(60);
	//separator = malloc(10);
	strcpy(fileName,"testData/sortTest");
	startTime = time(NULL);
	sprintf(fileName,"%s%d.txt",fileName,startTime);
	myFile = fopen(fileName,"w");
	printf("How many test cases to generate?");
	scanf("%ld",&numOfTestCases);
	printf("Maximum value?");
        scanf("%d",&max);
//	printf("Separator:");
//	separator[0] = fgetc(stdin);
	for (i=0;i<numOfTestCases;i++) {
		fprintf(myFile,"%d%s",(rand() % max) + 1,separator);
	}
	printf("%ld test cases generated and saved to file %s",numOfTestCases,fileName);
	fclose(myFile);
}
